File: TravellerCreator/ClientApp/src/app/models/sector-creator/race/races.ts
import {Race} from "./race";

export class Races {
  static Human: Race = {
    name: "humans",
    allegianceCode: "Hu",
    techLevel: 11,
    expansionLevel: 7,
    homeworldCoordinates: {column: 23, row: 11}
  };
  static Aslan: Race = {
    name: "aslan",
    allegianceCode: "As",
    techLevel: 10,
    expansionLevel: 6,
    homeworldCoordinates: {column: 17, row: 23}
  };
  static Mannu: Race = {
    name: "mannu",
    allegianceCode: "Ma",
    techLevel: 10,
    expansionLevel: 6,
    homeworldCoordinates: {column: 13, row: 8}
  };
  static Largosians: Race = {
    name: "largosians",
    allegianceCode: "La",
    techLevel: 10,
    expansionLevel: 7,
    homeworldCoordinates: {column: 17, row: 32}
  };
  static Tortosians: Race = {
    name: "tortosians",
    allegianceCode: "To",
    techLevel: 9,
    expansionLevel: 6,
    homeworldCoordinates: {column: 27, row: 31}
  };
  static Ithromir: Race = {
    name: "ithromir",
    allegianceCode: "It",
    techLevel: 13,
    expansionLevel: 8,
    homeworldCoordinates: {column: 8, row: 27}
  };
  static Chrotos: Race = {
    name: "chrotos",
    allegianceCode: "Ch",
    techLevel: 12,
    expansionLevel: 6,
    homeworldCoordinates: {column: 15, row: 17}
  };
  static TheCollective: Race = {
    name: "theCollective",
    allegianceCode: "Co",
    techLevel: 13,
    expansionLevel: 4,
    homeworldCoordinates: {column: 29, row: 18}
  };
  static KaSara: Race = {
    name: "kaSara",
    allegianceCode: "Ka",
    techLevel: 12,
    expansionLevel: 7,
    homeworldCoordinates: {column: 27, row: 2}
  };
  static Scrappers: Race = {
    name: "scrappers",
    allegianceCode: "Sc",
    techLevel: 15,
    expansionLevel: 2,
    homeworldCoordinates: {column: 5, row: 16}
  };
  static Vargr: Race = {
    name: "vargr",
    allegianceCode: "Vr",
    techLevel: 11,
    expansionLevel: 6,
    homeworldCoordinates: {column: 26, row: 24}
  } as Race;

  static AllRaces: Race[] = [this.Human, this.Aslan, this.Mannu, this.Largosians, this.Tortosians, this.Ithromir, this.Chrotos, this.TheCollective, this.KaSara, this.Scrappers, this.Vargr];
}

File: SectorCreator.Models/Basic/Races.cs
using SectorCreator.Global;

namespace SectorCreator.Models.Basic;

public static class Races
{
    private static readonly Race Human = new() {
        Name = "Human",
        HomeworldCoordinates = new Coordinates(18, 15),
        HomeSystemFilename = "../../../../Data/StarSystems/TerraSystem.tsv",
        Allegiance = "Humn",
        ExpansionRate = 8,
        Resilience = 1.25,
        MaxPopulation = 10
    };

    private static readonly Race Aslan = new() {
        Name = "Aslan",
        HomeworldCoordinates = new Coordinates(19, 6),
        HomeSystemFilename = "../../../../Data/StarSystems/KusyuSystem.tsv",
        Allegiance = "Asln",
        ExpansionRate = 8,
        Resilience = 1.25,
        MaxPopulation = 10
    };

    private static readonly Race Kekuu = new() {
        Name = "Kekuu",
        HomeworldCoordinates = new Coordinates(7, 32),
        HomeSystemFilename = "../../../../Data/StarSystems/KekuukiSystem.tsv",
        Allegiance = "Keku",
        ExpansionRate = 8,
        Resilience = .8,
        MaxPopulation = 9
    };

    private static readonly Race Ssitolusss = new() {
        Name = "Ssitolusss",
        HomeworldCoordinates = new Coordinates(13, 26),
        HomeSystemFilename = "../../../../Data/StarSystems/SsitolsssSystem.tsv",
        Allegiance = "Stls",
        ExpansionRate = 8,
        Resilience = .7,
        MaxPopulation = 9
    };

    private static readonly Race Tortosians = new() {
        Name = "Tortosians",
        HomeworldCoordinates = new Coordinates(19, 33),
        HomeSystemFilename = "../../../../Data/StarSystems/TortoriaSystem.tsv",
        Allegiance = "Tort",
        ExpansionRate = 7,
        Resilience = .85,
        MaxPopulation = 10
    };

    private static readonly Race Blubbus = new() {
        Name = "Blubbus",
        HomeworldCoordinates = new Coordinates(29, 29),
        HomeSystemFilename = "../../../../Data/StarSystems/BlubbusSystem.tsv",
        Allegiance = "Blbs",
        ExpansionRate = 10,
        Resilience = .65,
        MaxPopulation = 10
    };

    private static readonly Race Chrotos = new() {
        Name = "Chrotos",
        HomeworldCoordinates = new Coordinates(26, 14),
        HomeSystemFilename = "../../../../Data/StarSystems/ArianuilSystem.tsv",
        Allegiance = "Crts",
        ExpansionRate = 6,
        Resilience = .65,
        MaxPopulation = 8
    };

    private static readonly Race KaSara = new() {
        Name = "Ka'Sara",
        HomeworldCoordinates = new Coordinates(1, 10),
        HomeSystemFilename = "../../../../Data/StarSystems/KaTollisSystem.tsv",
        Allegiance = "KaSa",
        ExpansionRate = 16,
        Resilience = 3.5,
        MaxPopulation = 12
    };

    private static Race Vargr = new() {
        Name = "Vargr",
        HomeworldCoordinates = new Coordinates(26, 8),
        HomeSystemFilename = "../../../../Data/StarSystems/LairSystem.tsv",
        Allegiance = "Vrgr",
        ExpansionRate = 8,
        Resilience = 1.5,
        MaxPopulation = 10
    };

    public static readonly List<Race> races = new() {
        Human,
        Aslan,
        Kekuu,
        Ssitolusss,
        Tortosians,
        Blubbus,
        Chrotos,
        KaSara,
        Vargr
    };
}